File: Lovecraft/Api/Helper/StoryHelper.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text.RegularExpressions;

public class StoryHelper
{
    public List<string> SplitIntoPages(string text, int maxWordsPerPage)
    {
        string[] lines = text.Split(new[] { "\r\n", "\n" }, StringSplitOptions.None);

        List<string> pages = new List<string>();
        string currentPage = "";
        int wordCount = 0;

        foreach (string line in lines)
        {
            string[] words = line.Split(new[] { ' ' }, StringSplitOptions.RemoveEmptyEntries);

            foreach (string word in words)
            {
                currentPage += word + " ";
                wordCount++;

                if (wordCount >= maxWordsPerPage)
                {
                    pages.Add(currentPage.Trim());
                    currentPage = "";
                    wordCount = 0;
                }
            }
            currentPage += "</br>";
        }
        if (!string.IsNullOrEmpty(currentPage))
        {
            pages.Add(currentPage.Trim());
        }
        return pages;
    }
}

File: lovecraft-ts/src/helpers/story.helper.ts
import { IStoryDto, StoryWithRelations } from '../schemas/Story'
import { toUserDto } from '../helpers/user.helper'
import mammoth from "mammoth"

export const toStoryDto = (story: StoryWithRelations): IStoryDto => ({
  id: story.id,
  title: story.title,
  coverUrl: story.coverUrl,
  summary: story.summary,
  status: story.status,
  pages: story.pages,
  userId: story.userId,
  user: toUserDto(story.user),
  eventId: story.eventId,
  event: story.event,
  teamId: story.teamId,
  team: story.team,
  storyHistory: story.storyHistories ? story.storyHistories[0] : null,
  storyTags: null,
  storyVotes: story.storyVotes,
  createdAt: story.createdAt,
  updatedAt: story.updatedAt
})

export async function extractRawTextFromDocx(fileBuffer: Buffer): Promise<string> {
  const result = await mammoth.extractRawText({ buffer: fileBuffer })
  return result.value
}
export async function splitIntoPages(text: string, maxWordsPerPage: number): Promise<string[]> {
  const lines = text.split(/\r?\n/)

  const pages: string[] = []
  let currentPage = ""
  let wordCount = 0

  for (const line of lines) {
    const words = line.split(/\s+/).filter(w => w.length > 0)

    for (const word of words) {
      currentPage += word + " "
      wordCount++

      if (wordCount >= maxWordsPerPage) {
        pages.push(currentPage.trim())
        currentPage = ""
        wordCount = 0
      }
    }
    currentPage += "</br>"
  }

  if (currentPage.trim().length > 0) {
    pages.push(currentPage.trim())
  }

  return pages
}